### PR TITLE
[enhancemnt] support local up openyurt on mac machine

### LINKED ISSUE #811: [feature request] run `yurtctl test` on mac
**What would you like to be added**:
Ensure `yurtctl test` that can work correctly on Mac.

**Why is this needed**:
Currently, `yurtctl test` is ensured to run on the platform of `linux/amd64`, and other platforms have not been tested yet. Many developers may have their development environments based on Mac. It will be more convenient for them, if we can ensure `yurtctl test` to run correctly on Mac.

**others**
/kind feature

### PR DESCRIPTION
**Pull request by rambohe-ch**
/kind enhancement

#### What this PR does / why we need it:
In order to make OpenYurt developer can debug easily, In this pull request, we support to startup OpenYurt cluster locally on mac env. and the enhancements are as following:
1. use `docker buildx build` to build OpenYurt Components images that can run across multi platforms.
2. add `node-servant config control-plane` command for configuring kube-apiserver and kube-controller-manager static pod.
3. add `--enable-dummy-if` and `--enable-node-pool` parameter for `node-servant convert` command in order to disable dummy interface and node pool list/watch for yurthub in mac env
4. disable iptables mode for yurt-tunnel-server component
5. add `ConfigureAddons` process for setting coredns and kube-proxy pod.

#### Which issue(s) this PR fixes:

Fixes #811 

#### Does this PR introduce a user-facing change?

```release-note
// Startup OpenYurt Cluster on Linux/amd64: 
$# make local_up_openyurt TARGET_PLATFORMS=linux/amd64

// Startup OpenYurt Cluster on Linux/arm64(MACBook Pro M1): 
$# make local_up_openyurt TARGET_PLATFORMS=linux/arm64 ENABLE_DUMMY_IF=false
```

### CODE REVIEW

**Comment by Congrool** [comment]
review seems to need a little time to finish. I'll continue to review in the coming few days

File: Makefile [hunk 1]
```diff
@@ -12,7 +12,13 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-.PHONY: clean all release build
+TARGET_PLATFORMS ?= linux/amd64,linux/arm64,linux/arm/v7
+IMAGE_REPO ?=openyurt
+IMAGE_TAG ?= $(shell git describe --abbrev=0 --tags)
+ENABLE_DUMMY_IF ?=true
+ENABLE_NODE_POOL ?=false
```
[the remark this answers is not in the document]
Comment (rambohe-ch) [reply Makefile:19]: it's a good idea to set these two variables automatically. i will fix it.

[the remark this answers is not in the document]
Comment (rambohe-ch) [reply Makefile]: fixed

File: pkg/yurtctl/cmd/yurttest/kindinit/init.go [hunk 2]
```diff
@@ -242,77 +262,98 @@ func newKindInitializer(cfg *initializerConfig) *Initializer {
...
-		i.YurtControllerManagerImage,
-		i.NodeServantImage,
-		i.YurtTunnelServerImage,
-	}, i.CloudNodes); err != nil {
+	if err := ki.loadImagesToKindNodes([]string{
+		ki.YurtHubImage,
+		ki.YurtControllerManagerImage,
+		ki.NodeServantImage,
+		ki.YurtTunnelServerImage,
+		ki.NodeServantImage,
```
Comment (Congrool) [inline pkg/yurtctl/cmd/yurttest/kindinit/init.go:323]: `NodeServantImage` appears twice

File: pkg/node-servant/config/control-plane.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+type Runner interface {
+	Do() error
+}
+
+func NewControlPlaneRunner(o *ControlPlaneOptions) (Runner, error) {
+	switch o.RunMode {
+	case "pod":
+		return newStaticPodRunner(o.PodManifestsPath)
+	default:
```
[the remark this answers is not in the document]
Comment (rambohe-ch) [reply pkg/node-servant/config/control-plane.go:44]: when kube-apiserver and kube-controller-manager are deployed by Deployment controller, controller mode will be used. and controller mode will be implemented in the future.

File: cmd/yurt-node-servant/config/config.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+)
+
+// NewConfigCmd generates a new config command
+func NewConfigCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "config",
+		Short: "manage configuration of OpenYurt cluster",
+		RunE:  cmdutil.SubCmdRunE("config"),
+		Args:  cobra.NoArgs,
+	}
```
[the remark this answers is not in the document]
Comment (rambohe-ch) [reply cmd/yurt-node-servant/config/config.go:38]: fixed it.

File: pkg/node-servant/convert/options.go [hunk 5]
```diff
@@ -29,71 +28,54 @@ import (
...
-	}
-	o.yurthubImage = yurthubImage
-
-	yurthubHealthCheckTimeout, err := flags.GetDuration("yurthub-healthcheck-timeout")
-	if err != nil {
-		return err
+// Validate validates Options
+func (o *Options) Validate() error {
+	if len(o.joinToken) == 0 {
+		return fmt.Errorf("join token(bootstrap token) is empty")
```
[the remark this answers is not in the document]
Comment (rambohe-ch) [reply pkg/node-servant/convert/options.go:65]: a good idea. fixed it.

**Review by Congrool — COMMENTED** [review]
Other revise is at https://github.com/Congrool/openyurt/commit/ce245a89d98ca3fa2d60aaaf499d2dbd976ef650, including 
1. enable REGION, when set as cn, the GOPROXY and apk repository source will be set as a mirror in China.
2. decouple local-up-openyurt with docker-build, we don't need to always build the image before local-up-openyurt
3.  change the image-tag. For HEAD which is actually points at a tag, like v0.6.0, the image tag will be v0.6.0, otherwise the image tag will be v0.6.0-${commit}. Use this image tag, we can avoid overwriting the image which already has the tag v0.6.0 when building images.

File: hack/dockerfiles/Dockerfile.yurt-tunnel-server [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+FROM --platform=${BUILDPLATFORM} golang:1.17.1 as builder
+ADD . /build
+ARG TARGETOS TARGETARCH GIT_VERSION
+WORKDIR /build/
+RUN CGO_ENABLED=0 GOOS=${TARGETOS} GOARCH=${TARGETARCH} GIT_VERSION=${GIT_VERSION} make build WHAT=cmd/yurt-tunnel-server
+
+FROM --platform=${TARGETPLATFORM} alpine:3.14
+ARG TARGETOS TARGETARCH
+RUN apk add ca-certificates bash libc6-compat && update-ca-certificates && rm /var/cache/apk/*
```
Comment (Congrool) [inline hack/dockerfiles/Dockerfile.yurt-tunnel-server:11]: we should also `apk add iptables ip6tables` for yurt-tunnel-server.

File: hack/dockerfiles/Dockerfile.yurthub [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+
+FROM --platform=${BUILDPLATFORM} golang:1.17.1 as builder
+ADD . /build
+ARG TARGETOS TARGETARCH GIT_VERSION
+WORKDIR /build/
+RUN CGO_ENABLED=0 GOOS=${TARGETOS} GOARCH=${TARGETARCH} GIT_VERSION=${GIT_VERSION} make build WHAT=cmd/yurthub
+
+FROM --platform=${TARGETPLATFORM} alpine:3.14
+ARG TARGETOS TARGETARCH
+RUN apk add ca-certificates bash libc6-compat iptables ip6tables && update-ca-certificates && rm /var/cache/apk/*
```
Comment (Congrool) [inline hack/dockerfiles/Dockerfile.yurthub:11]: we should also `apk add iptables ip6tables` for yurthub, when enable-dummy-if is false, it needs to set iptables for dummy interface.

File: pkg/yurtctl/cmd/yurttest/kindinit/init.go [hunk 8]
```diff
@@ -327,29 +367,208 @@ func (i *Initializer) prepareKindConfigFile(kindConfigPath string) error {
...
+	}
+
+	// add annotation(openyurt.io/topologyKeys=kubernetes.io/hostname) for service kube-system/kube-dns in order to use the
+	// local coredns instance for resolving.
+	svc, err := ki.kubeClient.CoreV1().Services("kube-system").Get(context.TODO(), "kube-dns", metav1.GetOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+	if svc != nil && len(svc.Annotations["openyurt.io/topologyKeys"]) == 0 {
+		svc.Annotations["openyurt.io/topologyKeys"] = "kubernetes.io/hostname"
```
[the remark this answers is not in the document]
Comment (rambohe-ch) [reply pkg/yurtctl/cmd/yurttest/kindinit/init.go:463]: fixed it.

File: pkg/yurtctl/cmd/yurttest/kindinit/init.go [hunk 9]
```diff
@@ -327,29 +367,208 @@ func (i *Initializer) prepareKindConfigFile(kindConfigPath string) error {
...
+			klog.Info("coredns deployment configuration is completed")
+			return nil
+		}
+	}
+}
+
+func (ki *Initializer) configureCoreDnsAddon() error {
+	// config configmap kube-system/coredns in order to add hosts setting for resolving hostname to x-tunnel-server-internal-svc service
+	cm, err := ki.kubeClient.CoreV1().ConfigMaps("kube-system").Get(context.TODO(), "coredns", metav1.GetOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
```
[the remark this answers is not in the document]
Comment (rambohe-ch) [reply pkg/yurtctl/cmd/yurttest/kindinit/init.go:436]: fixed it.

File: hack/dockerfiles/Dockerfile.yurt-tunnel-server (hunk 6 above)
Comment (rambohe-ch) [reply hack/dockerfiles/Dockerfile.yurt-tunnel-server:11]: fixed it.

**Review by Congrool — APPROVED** [review]

**Review by DrmagicE — APPROVED** [review]

**Pull request merged**